src/ProductView/Diagram/LineChart.js

```diff
@@ -0,0 +1,90 @@
+import React, { Component } from 'react';
+import Chart from "chart.js";
+import {argosConfig} from '../../config/argosConfig.js';
+
+class LineChart extends Component {
+    
+    static sortEventsByTime(elementA, elementB, timeStampAttribute) {
+        let datePartsA = elementA[timeStampAttribute].split("T")[0];
+        let datePartsB = elementB[timeStampAttribute].split("T")[0];
+        datePartsA = datePartsA.split("-");
+        datePartsB = datePartsB.split("-");
+        const dateA = new Date(datePartsA[0], datePartsA[1], datePartsA[2]);
+        const dateB = new Date(datePartsB[0], datePartsB[1], datePartsB[2]);
+        return dateA - dateB;
+    }
+    
+    buildChartData() {
+        const dataset = [];
+        this.props.eventData.forEach((eventDataContainer) => {
+            dataset.push(this.buildChartDataset(eventDataContainer.eventType, eventDataContainer.events));
+        });
+        return dataset;
+    }
+    buildChartDataset(eventType, events) {
+        const dataContainer = [];
+        const timeStampAttribute = eventType.timestampAttributeName;
+        let counter = 0;
+        const sortedEvents = events;
+        sortedEvents.sort((elementA, elementB) => {
+            return LineChart.sortEventsByTime(elementA, elementB, timeStampAttribute);
+        });
+        sortedEvents.forEach(function (event) {
+            counter = counter + 1;
+            dataContainer.push({
+                x: event[timeStampAttribute].split("T")[0],
```

Comment:
The exact same splitting was used in LineChart > sortEventsByTime in line 9. Please extract it to a method.